utils/docker/Dockerfile.centos.8

```diff
@@ -105,11 +105,20 @@ gpgcheck=False\n" >> /etc/yum.repos.d/$repo:$branch:$build_number.repo;   \
         dnf repolist; \
         dnf --disablerepo=\* --enablerepo=$repo:$branch:$build_number makecache; \
     done
+# For conditional build DAOS and Java
+ARG DAOS_BUILD=$DAOS_DEPS_BUILD
+ARG DAOS_JAVA_BUILD=$DAOS_BUILD
+ARG BASE_DISTRO
```

Comment:
Putting it this early in the dockerfile will mean that changing if java is built or not will re-install all the packages and rebuild all the dependencies, it would be better to have the customisations (or at least the DAOS_JAVA_BUILD ones) later on, preferably after L245.